packages/models-library/src/models_library/groups.py

```diff
@@ -1 +1,47 @@
-EVERYONE_GROUP_ID = 1
+import enum
+from typing import Any, ClassVar, Final
+
+from models_library.utils.common_validators import create_enums_pre_validator
+from pydantic import BaseModel, Field, validator
+from pydantic.types import PositiveInt
+
+EVERYONE_GROUP_ID: Final[int] = 1
+
+
+class GroupTypeInModel(str, enum.Enum):  # noqa: SLOT000
+    """
+    standard: standard group, e.g. any group that is not a primary group or special group such as the everyone group
+    primary: primary group, e.g. the primary group is the user own defined group that typically only contain the user (same as in linux)
+    everyone: the only group for all users
+    """
+
+    STANDARD = "standard"
+    PRIMARY = "primary"
+    EVERYONE = "everyone"
+
+
+class Group(BaseModel):
+    gid: PositiveInt
+    name: str
+    description: str
+    group_type: GroupTypeInModel = Field(..., alias="type")
+    thumbnail: str | None
+
+    _from_equivalent_enums = validator("group_type", allow_reuse=True, pre=True)(
+        create_enums_pre_validator(GroupTypeInModel)
+    )
```

Comment:
Reading it again, if possible I would prefer the style we use everywhere.

Reply:
see above